src/mem/alloc.h

```diff
@@ -875,6 +875,60 @@ namespace snmalloc
 #endif
     }
 
+    /**
+     * If result parameter is non-null, then is_empty is passsed into the
+     * the location pointed to by result.
+     *
+     * Otherwise, asserts that it is empty.
+     **/
+    void debug_is_empty(bool* result)
+    {
+      auto test = [&result](bool value) {
```

Comment:
Rather than making this `bool value`, it would probably be better to make it `auto &allocator` and then do the `.is_empty()` call here and be able to report `&allocator` in the error message (or can we not do printf-like things here?).

Reply:
The Pal doesn't support printf like things, yet.